Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProjectTests/CourseAPIServiceTests/CourseAPIServiceTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  NetworkManagerTests.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 8/20/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import XCTest
+@testable import raywenderlich
+
+class CourseAPIServiceTests: XCTestCase {
+  var artworkUrl: URL?
+  var courseAPIService = CourseAPIService()
+
+  override func setUpWithError() throws {
+    artworkUrl = URL(string: "https://files.betamax.raywenderlich.com/attachments/collections/244/e2d20345-06fb-42f4-8366-c7ffe80b5f77.png")
+  }
```

Comment:
I appreciate that you wrote tests, but connecting to the internet from your tests is probably a bad idea.

1. It makes the tests slow. When you have hundreds of test methods, speed is very important.
2. You are essentially DDOS-ing your server.
3. It makes the tests depend on your server, internet connection and other factors outside your code. If your code works, your tests _shouldn't_ fail, even if the server has failed.
4. The tests are now very fragile since a change on the server could cause your tests to fail.

Instead, read up on dependency inversion and injection and how it relates to testing, and try to create a mock object that will return a static JSON file for the request, the JSON can be hard-coded inside your app. That way your tests become independent from the Internet.